lib/httparty/request.rb

```diff
@@ -78,7 +78,11 @@ def uri
 
       # avoid double query string on redirects [#12]
       unless redirect
-        new_uri.query = query_string(new_uri)
+        possible_query_string = query_string(new_uri)
+
+        if possible_query_string != nil && possible_query_string != ""
```

Comment:
Well… I was doing baby steps and the change was required, but I double-checked and it is not required anymore :)